crates/wdl-engine/src/backend/lsf_apptainer.rs

```diff
@@ -0,0 +1,648 @@
+                container,
+                cpu,
+                memory,
+                crankshaft_events: self.crankshaft_events.clone(),
+                cancellation_token,
+            },
+            completed_tx,
+        );
+
+        Ok(completed_rx)
+    }
+
+    fn cleanup<'a>(
+        &'a self,
+        _work_dir: &'a EvaluationPath,
+        _token: CancellationToken,
+    ) -> Option<futures::future::BoxFuture<'a, ()>> {
+        // TODO ACF 2025-09-11: determine whether we need cleanup logic here;
+        // Apptainer's security model is fairly different from Docker so
+        // uid/gids on files shouldn't be as much of an issue, and using only
+        // `apptainer exec` means no longer-running containers to tear down
+        None
+    }
+}
+
+/// Configuration for the LSF + Apptainer backend.
+// TODO ACF 2025-09-12: add queue option for short tasks
+//
+// TODO ACF 2025-09-23: add a Apptainer/Singularity mode config that switches around executable
+// name, env var names, etc.
+#[derive(Debug, Clone, serde::Deserialize, serde::Serialize)]
+pub struct LsfApptainerBackendConfig {
+    /// Which queue, if any, to specify when submitting jobs to LSF.
+    pub queue: Option<String>,
+    /// The maximum number of scatter subtasks that can be evaluated
+    /// concurrently.
+    ///
+    /// By default, this is 200.
+    #[serde(default = "default_max_scatter_concurrency")]
+    pub max_scatter_concurrency: u64,
+    /// The directory in which temporary directories will be created containing
+    /// Apptainer `.sif` files.
+    ///
+    /// This should be a location that is accessible by all jobs on the LSF
+    /// cluster.
+    ///
+    /// By default, this is `~/.cache/sprocket-apptainer-images`.
+    #[serde(default = "default_apptainer_images_dir")]
+    pub apptainer_images_dir: PathBuf,
+}
+
+fn default_max_scatter_concurrency() -> u64 {
+    200
+}
+
+fn default_apptainer_images_dir() -> PathBuf {
+    PathBuf::from("~/.cache/sprocket-apptainer-images")
+}
+
+impl Default for LsfApptainerBackendConfig {
+    fn default() -> Self {
+        Self {
+            queue: None,
+            max_scatter_concurrency: default_max_scatter_concurrency(),
+            apptainer_images_dir: default_apptainer_images_dir(),
+        }
+    }
+}
+
+impl LsfApptainerBackendConfig {
+    /// Validate that the backend is appropriately configured.
+    pub fn validate(&self, engine_config: &Config) -> Result<(), anyhow::Error> {
+        if cfg!(not(unix)) {
+            bail!("LSF + Apptainer backend is not supported on non-unix platforms");
+        }
+        if !engine_config.experimental_features_enabled {
+            bail!("LSF + Apptainer backend requires enabling experimental features");
+        }
+        // TODO ACF 2025-09-12: what meaningful work to be done here? Maybe ensure the
+        // queue exists, interrogate the queue for limits and match them up
+        // against prospective future config options here?
```

Comment:
Another option is to look for `Job not submitted` in the bsub stdout during the above submission logic. The queue name, specifically, could be validated here, but it might make sense to have a more general check that the job submitted successfully. 

```
$ bsub -K -q my_queue bash
HPCF: WARNING! No Memory was requested!
      A default memory request of 2.50 GB has been placed for this job
      The job will be killed if   2.50 GB of memory is used
my_queue: No such queue. Job not submitted.
```

Reply:
Looks like `bqueues -l <queue_name>` will do the trick too without actually submitting a job. Whether or not that's desirable is another question...

Reply:
I am going to punt this to later, since doing it properly implies doing blocking operations in `validate()`. We'll either have to make it `async` to bubble out the blocking explicitly, or denote the behavior and carefully wrap async calling contexts to avoid scheduler starvation.